ops/scripts/system_heartbeat_check.sh

```
@@ -0,0 +1,103 @@
+last_mini_restarts="0"
+if [[ -f "$ALERT_STATE_FILE" ]]; then
+  # shellcheck disable=SC1090
+  source "$ALERT_STATE_FILE" || true
+fi
+if [[ "${mini_restarts:-0}" =~ ^[0-9]+$ ]] && [[ "${last_mini_restarts:-0}" =~ ^[0-9]+$ ]]; then
+  if (( mini_restarts > last_mini_restarts )); then
+    issues+=("mini-api restart counter increased: ${last_mini_restarts} -> ${mini_restarts}")
+  fi
+fi
+
+latest_backup_ts="$(ls -1t "$BACKUP_DIR"/*.sql.gz 2>/dev/null | awk 'NR==1{print $1}')"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)